### ISSUE TITLE
MaterialRadio and other components break project

### ISSUE THREAD

**Issue by thosakwe** [dart-lang/angular2_components#31]
Hi. Upon upgrading from Angular 2.1.1 to Angular 2.2.0, my project instantly broke. Here is the otuput from attempting to load the `angular2` transformer:

```
[Error from DirectiveMetadataLinker on angular2_components|lib/src/components/material_radio/material_radio.ng_summary.json with input angular2_components|lib/src/components/material_radio/material_radio.ng_meta.json]:
Missing identifier "MaterialRadioGroupComponent" needed by "MaterialRadioComponent" from metadata map
[Error from DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json with input angular2_components|lib/angular2_components.ng_meta.json]:
Missing identifier "MaterialExpansionPanelSet" needed by "root" from metadata map
[Error from DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json with input angular2_components|lib/angular2_components.ng_meta.json]:
Missing identifier "MaterialRadioGroupComponent" needed by "root" from metadata map
[Error from DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json with input angular2_components|lib/angular2_components.ng_meta.json]:
Missing identifier "ReorderItemDirective" needed by "root" from metadata map
[Error from DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json with input angular2_components|lib/angular2_components.ng_meta.json]:
Missing identifier "ReorderListComponent" needed by "root" from metadata map
Multiple errors occurred:

- Transform DirectiveProcessor on angular2_components|lib/src/components/material_expansionpanel/material_expansionpanel_set.dart threw error: Queries inside constructors are deprecated, please replace with query annotations on fields in class MaterialExpansionPanelSet
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal/iterable.dart 416                                        MappedListIterable.elementAt
  dart:_internal/iterable.dart 222                                        ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  dart:async/future.dart 144                                              Future.Future.microtask.<fn>
  dart:async/zone.dart 1150                                               _rootRun
  dart:async/zone.dart 1026                                               _CustomZone.run
  dart:async/zone.dart 953                                                _CustomZone.bindCallback.<fn>
  dart:async/schedule_microtask.dart 41                                   _microtaskLoop
  dart:async/schedule_microtask.dart 50                                   _startMicrotaskLoop
  dart:isolate-patch/isolate_patch.dart 96                                _runPendingImmediateCallback
  dart:isolate-patch/isolate_patch.dart 149                               _RawReceivePortImpl._handleMessage
  
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal                                                          ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  
- Transform DirectiveProcessor on angular2_components|lib/src/components/material_radio/material_radio_group.dart threw error: Queries inside constructors are deprecated, please replace with query annotations on fields in class MaterialRadioGroupComponent
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal/iterable.dart 416                                        MappedListIterable.elementAt
  dart:_internal/iterable.dart 222                                        ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  dart:async/future.dart 144                                              Future.Future.microtask.<fn>
  dart:async/zone.dart 1150                                               _rootRun
  dart:async/zone.dart 1026                                               _CustomZone.run
  dart:async/zone.dart 953                                                _CustomZone.bindCallback.<fn>
  dart:async/schedule_microtask.dart 41                                   _microtaskLoop
  dart:async/schedule_microtask.dart 50                                   _startMicrotaskLoop
  dart:isolate-patch/isolate_patch.dart 96                                _runPendingImmediateCallback
  dart:isolate-patch/isolate_patch.dart 149                               _RawReceivePortImpl._handleMessage
  
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal                                                          ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  
- Transform DirectiveProcessor on angular2_components|lib/src/components/reorder_list/reorder_list.dart threw error: Queries inside constructors are deprecated, please replace with query annotations on fields in class ReorderListComponent
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal/iterable.dart 416                                        MappedListIterable.elementAt
  dart:_internal/iterable.dart 222                                        ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  dart:async/future.dart 144                                              Future.Future.microtask.<fn>
  dart:async/zone.dart 1150                                               _rootRun
  dart:async/zone.dart 1026                                               _CustomZone.run
  dart:async/zone.dart 953                                                _CustomZone.bindCallback.<fn>
  dart:async/schedule_microtask.dart 41                                   _microtaskLoop
  dart:async/schedule_microtask.dart 50                                   _startMicrotaskLoop
  dart:isolate-patch/isolate_patch.dart 96                                _runPendingImmediateCallback
  dart:isolate-patch/isolate_patch.dart 149                               _RawReceivePortImpl._handleMessage
  
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal                                                          ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  
- Transform DirectiveProcessor on angular2_components|lib/src/components/scorecard/scoreboard.dart threw error: Queries inside constructors are deprecated, please replace with query annotations on fields in class ScoreboardComponent
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal/iterable.dart 416                                        MappedListIterable.elementAt
  dart:_internal/iterable.dart 222                                        ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  dart:async/future.dart 144                                              Future.Future.microtask.<fn>
  dart:async/zone.dart 1150                                               _rootRun
  dart:async/zone.dart 1026                                               _CustomZone.run
  dart:async/zone.dart 953                                                _CustomZone.bindCallback.<fn>
  dart:async/schedule_microtask.dart 41                                   _microtaskLoop
  dart:async/schedule_microtask.dart 50                                   _startMicrotaskLoop
  dart:isolate-patch/isolate_patch.dart 96                                _runPendingImmediateCallback
  dart:isolate-patch/isolate_patch.dart 149                               _RawReceivePortImpl._handleMessage
  
  package:angular2/src/transform/common/type_metadata_reader.dart 1280:3  _reportDeprecation
  package:angular2/src/transform/common/type_metadata_reader.dart 1256:7  _getCompileDiDependencyMetadata.<fn>
  dart:_internal                                                          ListIterable.toList
  package:angular2/src/transform/common/type_metadata_reader.dart 1270:6  _getCompileDiDependencyMetadata
  package:angular2/src/transform/common/type_metadata_reader.dart 229:13  _CompileTypeMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 673:10  _DirectiveMetadataVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:angular2/src/transform/common/type_metadata_reader.dart 72:10   TypeMetadataReader.readTypeMetadata
  package:angular2/src/transform/directive_processor/rewriter.dart 98:10  _NgMetaVisitor.visitClassDeclaration
  package:analyzer/src/dart/ast/ast.dart 1847:15                          ClassDeclarationImpl.accept
  package:analyzer/src/dart/ast/ast.dart 7688:20                          NodeListImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 92:30  _NgMetaVisitor.visitCompilationUnit
  package:analyzer/src/dart/ast/ast.dart 2439:15                          CompilationUnitImpl.accept
  package:angular2/src/transform/directive_processor/rewriter.dart 52:16  createNgMeta.<async>.<fn>.<async>
  
- Transform DirectiveMetadataLinker on angular2_components|lib/src/components/material_radio/material_radio.ng_summary.json threw error: Missing identifier "MaterialRadioGroupComponent" needed by "MaterialRadioComponent" from metadata map
- Transform DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json threw error: Missing identifier "MaterialExpansionPanelSet" needed by "root" from metadata map
- Transform DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json threw error: Missing identifier "MaterialRadioGroupComponent" needed by "root" from metadata map
- Transform DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json threw error: Missing identifier "ReorderItemDirective" needed by "root" from metadata map
- Transform DirectiveMetadataLinker on angular2_components|lib/angular2_components.ng_summary.json threw error: Missing identifier "ReorderListComponent" needed by "root" from metadata map
dart:isolate  _RawReceivePortImpl._handleMessage
```

**Comment by thosakwe** [comment]
Also, downgrading back to 2.1.1 produces the same error.

**Comment by nshahan** [comment]
Thanks. We are tracking this in issue #28. It should work again if you can ensure that Angular2 2.1.0 is being used. A fix is coming soon though.

**Comment by nshahan** [comment]
The fix is in #30 and we published angular2_components version 0.2.1 of the components. It is working with angular2 version 2.2.0 now.
